### ISSUE TITLE
disable missing space before function parentheses

### ISSUE THREAD

**Issue by kinghadi** [prettier/eslint-plugin-prettier#44]
Hi :)
First of all, thank u for this plugin.

**Problem**
For this piece of code:
```js
async function main({ host = HOST, port = PORT } = {}) { 
...
```
I get the following warning on my project:

![screenshot_2017-08-16_17-28-08](https://user-images.githubusercontent.com/3697216/29364377-75f08e88-82a8-11e7-807e-24c57cef2b85.png)

When I add a space according to the warning then I get an error by `prettier` that says remove the space. When I remove your plugin it doesn't warn me anymore but I have to add a space after the function name.
 
**How can I fix that? Are there any options that I can pass to disable this?**

**Environment**
I'm using 
 - Atom 1.19
 - Node 8
 - Linux OS
 
**.eslintrc.json**
```json
{
  "parser" : "babel-eslint",
  "extends": [
    "prettier"
  ],
  "plugins": [
    "prettier"
  ],
  "parserOptions": {
    "ecmaVersion": 2016,
    "sourceType": "module"
  },
  "env": {
    "es6": true,
    "node": true
  },
  "rules": {
    "curly": ["error", "all"],
    "max-len": ["error", {"code": 80, "ignoreUrls": true}],
    "no-confusing-arrow": "error",
    "no-tabs": "error",
    "quotes": ["error", "backtick"],
    "space-in-parens": 2,
    "space-before-function-paren": 0,
    "prettier/prettier": ["error",
      {
      "semi": false,
      "singleQuote": true
      }
    ]
  }
}
```

**package.json**
```json
...
"devDependencies": {
    "babel-cli": "^6.24.1",
    "babel-eslint": "^7.2.3",
    "babel-preset-env": "^1.6.0",
    "babel-preset-es2017": "^6.24.1",
    "eslint": "^4.4.1",
    "eslint-config-prettier": "^2.3.0",
    "eslint-plugin-prettier": "^2.2.0",
    "prettier": "^1.5.3"
  }
...
```

**Comment by not-an-aardvark** [comment]
This plugin works best if you disable all other ESLint rules relating to code formatting, and only enable rules that detect patterns in the AST. (If another active ESLint rule disagrees with `prettier` about how code should be formatted, it will be impossible to avoid lint errors.) You can use [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) to disable all formatting-related ESLint rules. If your desired formatting does not match the `prettier` output, you should use a different tool such as [prettier-eslint](https://github.com/prettier/prettier-eslint) instead.
